fetchChallenges.js

```diff
@@ -1,64 +1,65 @@
-const res = await fetch(
-  "https://lernia-sjj-assignments.vercel.app/api/challenges"
-);
-const data = await res.json();
-export const challenges = data.challenges;
-if (window.location.pathname.includes("ourChallenges")) {
-  const container = document.querySelector(".ourChallenges");
-  renderChallenges(container);
+const data = await fetchData(`https://lernia-sjj-assignments.vercel.app/api/challenges`).catch(error => console.log(error.message))
```

Comment:
Loading data immediately on the highest level (global scope) is problematic because it means you can't control when that happens. I understand that you want it to happen immediately, and that's why it's fine for this scenario. But it's good to write code that can run in different situations, specifically in the app and in tests. None of the code in this file can be unit tested because it can't be imported/loaded without trying to load data, which can cause tests to fail.

That's why we often abstract loading in a specialized class or function, that can be "injected" to the code you have here.